src/api.ts

```diff
@@ -90,13 +91,19 @@ export type CourseTopics = {
   items: string[];
 }[];
 
+export type DataCourseTopics = {
+  title: string;
+  items: { subtitle: string; subitems: string[] }[];
+}[];
+
 export type Course = {
   id: string;
   coach: string;
+  type: 'web' | 'java' | 'data';
   startDate: string;
   endDate: string;
   location: string;
-  topics: CourseTopics;
+  topics: CourseTopics | DataCourseTopics;
 };
```

Comment:
I guess you can not have a course with `type: "web"` and `topics` of type `DataCourseTopics`.
Thus, you could improve the typing here to allow only specific combinations.

As an example:

```ts
export type Course = {
  id: string;
  coach: string;
  startDate: string;
  endDate: string;
  location: string;
} & ({
  type: 'web' | 'java';
  topics: CourseTopics;
} | {
  type: 'data';
  topics: DataCourseTopics;
});
```

Feel free to create sub types like 
```ts
type DataCoursePartial = {
  type: 'data';
  topics: DataCourseTopics;
}
```